### PR TITLE
test: [ANDROAPP-3300] Search UI test

### PR DESCRIPTION
**Pull request by Sharmyn28**
## Description
This PR cotains tests such as:

- search by one field
- search by two fields
- error when search by one field
- change between programs
- search and check display list fields 

[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3300)

### CODE REVIEW

File: app/src/androidTest/java/org/dhis2/usescases/searchte/SearchTeiRobot.kt [hunk 1]
```diff
@@ -53,6 +56,13 @@ class SearchTeiRobot : BaseRobot() {
             )
     }
 
+    fun searchByField(searchWord: String, field: String) {
+        waitToDebounce(1000)
```
Comment (JaimeToca) [inline app/src/androidTest/java/org/dhis2/usescases/searchte/SearchTeiRobot.kt:60]: is this needed?

File: app/src/androidTest/java/org/dhis2/usescases/searchte/SearchTETest.kt [hunk 2]
```diff
@@ -3,36 +3,174 @@ package org.dhis2.usescases.searchte
...
-    fun openSearchTE() {
+    fun shouldSuccessfullySearchByName() {
+        val firstName = "Tim"
+        val firstNamePosition = 0
+        val filterCount = "1"
+
+        prepareChildProgrammeIntentAndLaunchActivity()
+
+        searchTeiRobot {
+            searchByPosition(firstName, firstNamePosition)
```
Comment (JaimeToca) [inline app/src/androidTest/java/org/dhis2/usescases/searchte/SearchTETest.kt:31]: Do you think _**typeAttribute(firstName, firstNamePosition)**_ or _**typeAttributeAtPosition(firstName, firstNamePosition)**_ are more accurate than "searchByPosition"

File: app/src/androidTest/java/org/dhis2/usescases/searchte/entity/DisplayListFieldsUIModel.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package org.dhis2.usescases.searchte.entity
+
+data class DisplayListFieldsUIModel(
+    val name: String,
+    val lastName: String,
+    val birthday: String,
+    val email: String,
+    val address: String,
+    val weight: String,
+    val height: String
```
Comment (JaimeToca) [inline app/src/androidTest/java/org/dhis2/usescases/searchte/entity/DisplayListFieldsUIModel.kt:10]: For now this is okay. If at some point we use a different program this attributes could change. If this happens in the future we could use Map<String, String> instead and make both value/key dynamic

File: app/src/androidTest/java/org/dhis2/usescases/searchte/SearchTETest.kt (hunk 2 above)
Comment (Sharmyn28) [reply app/src/androidTest/java/org/dhis2/usescases/searchte/SearchTETest.kt:31]: great I'll rename the function

Comment (JaimeToca) [reply app/src/androidTest/java/org/dhis2/usescases/searchte/SearchTETest.kt:31]: thanks @Sharmyn28

**Comment by JaimeToca** [comment]
Thanks @Sharmyn28 good job :) . Let me know when you change the names suggested !

**Review by andresmr — APPROVED** [review]

**Comment by andresmr** [comment]
Good job @Sharmyn28 !

**Review by Balcan — APPROVED** [review]

**Review by ferdyrod — APPROVED** [review]

**Review by mmmateos — APPROVED** [review]

**Pull request merged**